icu4c/source/common/bytesinkutil.cpp

```diff
@@ -121,35 +121,41 @@ ByteSinkUtil::appendUnchanged(const uint8_t *s, const uint8_t *limit,
     return TRUE;
 }
 
-CharStringByteSink::CharStringByteSink(CharString* dest, UErrorCode& status)
-        : dest_(dest), status_(status) {
-    if (U_SUCCESS(status_) && dest_ == nullptr) {
-        status_ = U_ILLEGAL_ARGUMENT_ERROR;
-    }
+CharStringByteSink::CharStringByteSink(CharString* dest) : dest_(*dest) {
 }
 
 CharStringByteSink::~CharStringByteSink() = default;
 
 void
 CharStringByteSink::Append(const char* bytes, int32_t n) {
-    if (U_SUCCESS(status_)) {
-        dest_->append(bytes, n, status_);
-    }
+    UErrorCode status;
+    dest_.append(bytes, n, status);
+    // Any errors are silently ignored.
```

Comment:
You are right, this swallows failures. In a follow-up change (to not block current work) we could add a flag or UErrorCode back into the class, just for Append(). I was looking at the other functions that should not fail.